src/components/info-tile/info-tile.tsx

```diff
@@ -0,0 +1,265 @@
+import { Component, Prop, h } from '@stencil/core';
+
+/**
+ * This component can be used on places such as a start page or a dashboard.
+ * It offers features for visualizing aggregated data along with supplementary
+ * information.
+ *
+ * Clicking on the component should navigate the user to
+ * a new screen or web page, to which you need to provide a URL,
+ * using the `href` property.
+ *
+ * By default, this navigation will happen within the same browser tab.
+ * However, it is possible to override that behavior,
+ * using the `target` property.
+ *
+ * @exampleComponent limel-example-info-tile
+ * @exampleComponent limel-example-info-tile-notification
+ * @exampleComponent limel-example-info-tile-progress
+ * @exampleComponent limel-example-info-tile-loading
+ * @exampleComponent limel-example-info-tile-styling
+ */
+
+@Component({
+    tag: 'limel-info-tile',
+    shadow: true,
+    styleUrl: 'info-tile.scss',
+})
+export class InfoTile {
+    /**
+     * Name of icon for the info-tile.
+     */
+    @Prop()
+    public icon?: string;
+
+    /**
+     * The text to show below the info-tile. Long label will be truncated.
+     */
+    @Prop({ reflect: true })
+    public label?: string = null;
+
+    /**
+     * A string of text that is visually placed before the value.
+     */
+    @Prop({ reflect: true })
+    public prefix?: string;
+
+    /**
+     * A piece of text or number that is the main piece of information
+     * which the component is intended to visualize.
+     */
+    @Prop({ reflect: true })
+    public value: number | string;
+
+    /**
+     * A string of text that is visually placed after the value.
+     */
+    @Prop({ reflect: true })
+    public suffix?: string;
+
+    /**
+     * The `title` tag of the hyperlink, which can be used to
+     * provide additional information about the link.
+     * It improves accessibility both for sighted users
+     * and users with assistive technologies.
+     */
+    @Prop({ reflect: true })
+    public linkTitle?: string = null;
+
+    /**
+     * Set to `true` if info-tile is disabled.
+     */
+    @Prop({ reflect: true })
+    public disabled: boolean = false;
+
+    /**
+     * The url that the info-tile leads to.
+     */
+    @Prop({ reflect: true })
+    public href?: string = null;
+
+    /**
+     * Set to `true` if info-tile is clickable.
+     */
+    @Prop({ reflect: true })
+    public clickable: boolean = false;
+
+    /**
+     * Where to load the linked URL, as the name for a browsing context:
+     * - `_self`: in the current browsing context. (Default)
+     * - `_blank`: in a new tab.
+     * - `_parent`: in the parent browsing context of the current one.
+     * If no parent, behaves as `_self`.
+     * - `_top`: the topmost browsing context (the "highest" context
+     * that's an ancestor of the current one). If no ancestors, behaves as `_self`.
+     */
+    @Prop({ reflect: true })
+    public target: '_self' | '_blank' | '_parent' | '_top' = '_self';
```

Comment:
I think we should combine `linkTitle`, `href` and `target` in to a single property, e.g. `link`
```ts
interface InfoTileLink {
    title: string;
    href: string;
    target: string;
}
```

Reply:
As I mentioned above, I think we should remove this functionality altogether and let the consumer handle anything that happens on click

Reply:
If the whole component should act as a single link, then yes, we could have the consumer wrap the whole thing in an `<a href="…">`.

But I'm not sure that's a very helpful approach towards the consumer. I generally agree about keeping components simple (or "stupid"), but they also need to be helpful enough to actually be valuable… 🤔 I'm not entirely sure where I stand on this yet…

Reply:
They would not have to wrap it in a link, the can just add an `onClick` handler depending on what the use case is. To me, the `clickable` prop is a strong indication that it's not clear how the component is supposed to be used and that it might be trying to do too much.

If we are not sure about the use case, I would say that we should start with the most simple solution (i.e. not having this functionality). We can always add it later if we find that we need it, it's a lot harder to remove it once we have introduced it.

Reply:
> They would not have to wrap it in a link, the can just add an `onClick` handler depending on what the use case is.

Yeah, but we really want to start fostering the use of proper links, in my opinion. We need to move away from this trend of not using proper links for things that should cause navigation.

> To me, the `clickable` prop is a strong indication that it's not clear how the component is supposed to be used and that it might be trying to do too much.

I agree. I think we could simply use the presence of a value for `href`, as you proposed originally. People should supply a url, and if they really can't, but still want to use a clickHandler, and have the "clickable" styling, they can just supply `'#'` as the url, and then handle the click in their own click handler.

Reply:
Also, if we speak from the use case of Lime CRM webclient this will be hard to use today since clicking an anchor tag will probably reload the whole page. I think we need something like the [Navigation API](https://developer.chrome.com/docs/web-platform/navigation-api/) to intercept navigation events like that, but I have not read the spec in detail and it's not even implemented everywhere yet.

Reply:
My quick take on it

- `limel-shortcut` is quite similar to this one, structurally and when it comes to the usages of links.
- The main idea here (for the web client) is that this is supposed to be used as a link.
- Lime Elements is living between the two worlds. But I think we should mostly care about how things are sed in the web client, and its eco system.
- During many previous conversations, have have come to the conclusion that we need to use `<a>` for navigations, and not for example `<button>`s or `<div class="trigger" tabindex="0"`. One of the reasons is giving ability to META-key + click, and open things in new browser tabs.
- Not sure how much we have to care about how the web client behaves with anchor links. In that case, that sounds like a buggy behavior of the client and shouldn't have anything to do with us here. If it's not possible to fix, that's another story.
- Links come with accessibility concerns and responsibilities. If we leave all of that to the consultants, they will make inaccessible links. This component tries to address those internally, as good as "I" personally can. So we don't bother documenting too much or risk consultants making too inaccessible solutions. These things are becoming laws, and we will have too many problems.
- People can add onClick, I don't think it's a super good idea. But they can and sometimes have to and will 🤷‍♂️ so why not take it into account.
- Start small, then make things bigger. I'm not a big fan honestly. I need these features right now, since I have to implement this component already.

Reply:
> Also, if we speak from the use case of Lime CRM webclient this will be hard to use today since clicking an anchor tag will probably reload the whole page. I think we need something like the [Navigation API](https://developer.chrome.com/docs/web-platform/navigation-api/) to intercept navigation events like that in order to push to history state, but I have not read the spec in detail and it's not even implemented everywhere yet.

I don't think we _need_ the Navigation API for that, although having it would probably make things easier.

Reply:
> I don't think we need the Navigation API for that, although having it would probably make things easier.

Yeah maybe we have some global click listener registered somewhere that can take care of it :smile:

Reply:
> * Start small, then make things bigger. I'm not a big fan honestly. I need these features right now, since I have to implement this component already. I'd rather make the component `@private` first, if we are worried for consequences.

My thumbs up and heart on your post refers to everything you said, _except_ this 😅

I have much to say on this, and I'm starting to feel like I'm maybe, possibly, _perhaps_, starting to get some thoughts formulated in a way that I might be able to convey them in a way that could, if I'm lucky, make sense to people outside my own head… maybe… 🙈

Let's have a huddle sometime…

Reply:
> I have much to say on this, 

I already know your standpoint on this. 😄 but a huddle would be nice. I also have a lot to say on this I guess. 

Regarding this particular component, features are based on current info-tile [^1]. I need to replace that for the multiple start pages to work smoothly. I don't wanna get stuck on technicalities, best practices, and bureaucracy for too long. We have already had a fair share of these types of discussions when I was trying to do this in the CRM-components repo.

I want move on.

But misunderstand me correctly now. 😸 I'm trying to say, please don't ask me to remove features to make the first PR small, and then gradually add more features, until I can bring it up to a level that I can use the component.

[^1]: and the needs that I am aware of, taking to others regarding other projects

Reply:
> I don't wanna get stuck on technicalities, best practices, and bureaucracy for too long. We have already had a fair share of these types of discussions when I was trying to do this in the CRM-components repo.
> 
> I want move on.

Sorry, but I do not agree at all. This is hardly technicalities nor bureaucracy. The public interface of the component is one of the most important aspects since it is what the consumer will use, so it's important that it's easy to use and makes sense. Once we have released something it can be extremely hard to remove or change without making a lot of people annoyed. That's why it's usually better to start small and add more stuff later as they are needed

Reply:
I already agree to that @jgroth 

> But misunderstand me correctly now. 😸 I'm trying to say, please don't ask me to remove features to make the first PR small, and then gradually add more features, until I can bring it up to a level that I can use the component.

This was a response or clarification to
> Start small, then make things bigger.

Reply:
> > I don't think we need the Navigation API for that, although having it would probably make things easier.
> 
> Yeah maybe we have some global click listener registered somewhere that can take care of it 😄

I might have a look at creating one today…

Reply:
Did we come to a conclusion about this? :smile: 
If you want to include the link functionality, I suggest grouping the props under a single prop named `link` like my initial comment. Also, I think we should remove the `clickable` prop and do what @adrianschmidt suggested, i.e. if the consumer want to have their own handler they have to add an empty `href` or similar and then implement their own `onClick`

Reply:
I was hoping you'd come to that conclusion 😁

This will give you the functionality you need @Kiarokh 👍

Reply:
As long as there is a way to get the interactive effects. The clickable was adding styles for mouse hover, mouse press, keyboard tab-focus, elevated shadow. 

https://user-images.githubusercontent.com/35954987/203765427-ab941a44-5c2b-4a76-b41a-9712d36f80d3.mov

I don't know how it works.

Can we automatically get these styles if there is an `onClick` defined from outside? 

In the video above, only one of them has a link. So all widgets look as if they are static.

Reply:
> Can we automatically get these styles if there is an `onClick` defined from outside?

Yes, the consumer just has to set `link={{ href: '#' }}`.

Reply:
The link properties has been combined to a single interface now. It's called `Link`, and there are probably other places we can use it too, like limel-shortcut for example 🙂

Reply:
> > Yes, the consumer just has to set link={{ href: '#' }}.
> 
> then we should perhaps document it? Can we reuse what I wrote about `isClickable`?

Fixed!

<img width="567" alt="" src="https://user-images.githubusercontent.com/154725/204269081-6bd9f2cb-b90a-43f9-8e59-84fc9c75b81f.png">

https://github.com/Lundalogik/lime-elements/pull/1911/files#diff-2b2e96118baabdc2489f06cff2a80821d813b0d72c449ae930a593a1f0e243c9R77

Reply:
I've squashed the fixups, and will mark this thread as resolved. I'll then proceed to approve and merge. I don't think anyone will mind, right? 😝